cmd/podman/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,238 @@
+
+		table := bpf.NewTable(m.TableId("events"), m)
+		channel := make(chan []byte)
+		perfMap, err := bpf.InitPerfMap(table, channel)
+		if err != nil {
+			log.Println("unable to init perf map")
+		}
+
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt, os.Kill)
+		rsc := false
+		go func() {
+			var e event
+			for {
+				data := <-channel
+				err := binary.Read(bytes.NewBuffer(data), binary.LittleEndian, &e)
+				if err != nil {
+					log.Printf("failed to decode received data '%s': %s\n", data, err)
+					continue
+				}
+				// comm := (*C.char)(unsafe.Pointer(&e.Command))
+				// log.Printf("Pid : %d, Syscall_ID : %d, , Command : %q", e.Pid, e.Id, C.GoString(comm))
+				name := getName(e.ID)
+				if name == "seccomp" {
+					rsc = true
+					continue
+				}
+				if rsc {
+					syscalls[name]++
+				}
+			}
+		}()
+		log.Println("PerfMap Start")
+		perfMap.Start()
+		<-sig
+		log.Println("PerfMap Stop")
+		perfMap.Stop()
+		generateProfile(syscalls, os.TempDir()+"/profile.json")
+	} else if *terminate {
+		log.Println("terminator")
+		f, err := ioutil.ReadFile("pid")
+
+		if err != nil {
+			log.Println(err)
+		}
+
+		log.Println("file opened")
+		Spid := string(f)
+
+		log.Println(Spid)
+		pid, _ := strconv.Atoi(Spid)
+		p, _ := os.FindProcess(pid)
+		p.Signal(os.Interrupt)
+
+	} else {
+		var s spec.State
+		reader := bufio.NewReader(os.Stdin)
+		decoder := json.NewDecoder(reader)
+		err := decoder.Decode(&s)
+		if err != nil {
+			log.Println(err)
+		}
+		pid := s.Pid
+		//sysproc := &syscall.SysProcAttr{Noctty: true}
+		attr := &os.ProcAttr{
+			Dir: ".",
+			Env: os.Environ(),
+			Files: []*os.File{
+				os.Stdin,
+				nil,
+				nil,
+			},
+			// Sys: sysproc,
+		}
+		if pid > 0 {
+			process, err := os.StartProcess("/usr/libexec/oci/hooks.d/oci-trace-hook", []string{"/usr/libexec/oci/hooks.d/trace", "-r", strconv.Itoa(pid)}, attr)
+			if err != nil {
+				log.Println("cannot launch process ", err.Error())
+				return
+			}
+			time.Sleep(2 * time.Second)
+			processPID := process.Pid
+			pwd, _ := os.Getwd()
+			log.Println("Currently in the directory ", pwd)
+			f, err := os.Create("pid")
+			if err != nil {
+				log.Println("Cannot write pid to file")
+			}
+			defer f.Close()
+			f.WriteString(strconv.Itoa(processPID))
+			err = process.Release()
+			if err != nil {
+				log.Println("cannot detach process", err.Error())
+			}
+			log.Println("PID of the process : ", processPID)
+		}
+	}
+
+}
+
+func generateProfile(c calls, fileName string) {
+	s := types.Seccomp{}
+	var names []string
+	for s, t := range c {
+		if t > 0 {
+			names = append(names, s)
+		}
+	}
+	s.DefaultAction = types.ActErrno
+
+	s.Syscalls = []*types.Syscall{
+		&types.Syscall{
+			Action: types.ActAllow,
+			Names:  names,
+			Args:   []*types.Arg{},
+		},
+	}
+	sJSON, _ := json.Marshal(s)
+
+	err := ioutil.WriteFile(fileName, sJSON, 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getName(id uint32) string {
```

Comment:
I think you want to handle the error here, or else you potentially end up with empty string syscalls.